gfx-examples/src/shadow.rs

```diff
@@ -658,28 +664,42 @@ impl framework::Example for Example {
     }
 
     fn render(&mut self, frame: &wgpu::SwapChainOutput, device: &mut wgpu::Device) {
-        for entity in &mut self.entities {
-            if entity.rotation_speed != 0.0 {
-                let rotation = cgmath::Matrix4::from_angle_x(cgmath::Deg(entity.rotation_speed));
-                entity.mx_world = entity.mx_world * rotation;
+        let mut encoder = device.create_command_encoder(&wgpu::CommandEncoderDescriptor { todo: 0 });
+
+        {
+            let size = mem::size_of::<EntityUniforms>() as u32;
+            let entities = self.entities.iter_mut().map(|mut entity| {
+                if entity.rotation_speed != 0.0 {
+                    let rotation = cgmath::Matrix4::from_angle_x(cgmath::Deg(entity.rotation_speed));
+                    entity.mx_world = entity.mx_world * rotation;
+                }
+                EntityUniforms {
+                    model: entity.mx_world.clone(),
+                    color: [entity.color.r, entity.color.g, entity.color.b, entity.color.a],
+                }
+            }).collect::<Vec<_>>();
```

Comment:
would be good to show how the mapping is filled up without intermediate `Vec` collected